packages/components/src/components/badge/badge.ts

```diff
@@ -26,29 +26,42 @@ export default class SdBadge extends SolidElement {
   /** Inverts the badge. */
   @property({ type: Boolean, reflect: true }) inverted = false;
 
+  private getParentElement() {
+    const parent = this.parentElement;
+    const isInteractiveElement = parent?.matches(
+      'a, button, sd-button, input, textarea, select, details, [href], [tabindex], [role="button"], [role="link"]'
+    );
+
+    return isInteractiveElement ? parent : null;
+  }
+
   render() {
+    const parentIsInteractive = !!this.getParentElement();
+
     return html`
       <span
-        tabindex="0"
+        tabindex=${parentIsInteractive ? nothing : '0'}
```

Comment:
I would remove the tabindex completely. I don't see a reason why this should be focusable at all.

Reply:
Updated.